### ISSUE TITLE
WS RPC connection isn't dropped on error

### ISSUE THREAD

**Issue by nazar-pc** [paritytech/substrate#10672]
Attempt to send something too big over WS RPC connection results in this:
```
WS Error <Capacity>: Reached the limit of the output buffer for the connection.
```

After which as far as I can tell RPC connection is dead, but it is not dropped by Substrate node.

I think it really should be dropped once such error happens since connection is effectively broken and useless.

**Comment by bkchr** [comment]
Can you give an example on how to trigger this?

**Comment by nazar-pc** [comment]
Take any existing RPC endpoint that returns a vector of bytes, change it to return `vec![0u8; 1024 * 1024 * 10]` (will overflow default output buffer with any serialization method used) and make request to that endpoint with any RPC client (I used `wscat` for simplicity).

Once endpoint is triggered and above error is printed, node stops responding to any subsequent messages sent over still open WebSocket connection and connection just idles forever after that.

**Comment by bkchr** [comment]
CC @niklasad1 @dvdplm

**Comment by niklasad1** [comment]
Ok, I think this is may https://github.com/paritytech/ws-rs/pull/8 that you run into.

We are in the process of replacing the jsonrpc rpc server with jsonrpsee I hope [this will merged "soon"](https://github.com/paritytech/substrate/pull/8783), so I hope this will be fixed by that (ideally we won't have to debug and fix this is ws-rs)

However, meanwhile we would really appreciate if you could try out the jsonrpsee branch but it would require some changes I could probably help you out if you want.

**Comment by nazar-pc** [comment]
On `dp-jsonrpsee-integration-2` branch it fails on serialization step, thus not corrupting connection in the first place:
> Error serializing response: Error("Memory capacity exceeded", line: 0, column: 0)

Also client receives an error in response:
> {"jsonrpc":"2.0","error":{"code":-32702,"message":"Response is too big","data":"Exceeded max limit 15728640"},"id":1}

So yeah, it does fix the problem, but PR is open since May of last year, so having a fix in the meantime can still be valuable.

There could be other cases where buffer might be overflown, but I have no easy way to test it, so we can worry about it if it ends up being a problem.

**Comment by niklasad1** [comment]
Ok, cool.

I expect it to merged soon (within a couple weeks) and it's not really fixable without patching `ws-rs` again. `ws-rs` is really complicated with the recycling/re-use of cyclic buffers and so on so you might be right. So it's not a priority right now (sorry) I hope we can abandon it completely

**Comment by niklasad1** [comment]
This should be closed by #8783, please open this again if you encounter this again.
